src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/Worker.java

```diff
@@ -651,14 +675,21 @@ private void initialize() {
                     } else {
                         LOG.info("LeaseCoordinator is already running. No need to start it.");
                     }
-                    isDone = true;
+                    LOG.info("Checking the shardSyncStrategy");
+                    shardSyncStrategy = shardSyncStrategyDecider.getShardSyncStrategy();
+                    shardSyncStrategy.onWorkerInitialization();
```

Comment:
You should move this inside the conditional that confirms that `shardSyncStrategy` is non-null.

Reply:
getShardSyncStrategy() could return null if the strategy is not able to be determined, in which case isDone is not set to true so that we have another initialization attempt

Reply:
If `shardSyncStrategy` is `null`, then line 680 will throw an NPE. Let's avoid that.